scipy/optimize/_highs/setup.py

```diff
@@ -0,0 +1,163 @@
+        start_idx = s.find(start_token) + len(start_token) + 1
+        end_idx = s[start_idx:].find(end_token) + len(s[:start_idx])
+    return s[start_idx:end_idx].strip()
+
+
+def configuration(parent_package='', top_path=None):
+
+    from numpy.distutils.misc_util import Configuration
+    config = Configuration('_highs', parent_package, top_path)
+
+    # HiGHS info
+    _major_dot_minor = _get_version('CMakeLists.txt', 'project(HIGHS VERSION', 'LANGUAGES CXX C')
+    HIGHS_VERSION_MAJOR, HIGHS_VERSION_MINOR = _major_dot_minor.split('.')
+    HIGHS_VERSION_PATCH = _get_version('CMakeLists.txt', 'HIGHS_VERSION_PATCH')
+    GITHASH = 'n/a'
+    HIGHS_DIR = str(pathlib.Path(__file__).parent.resolve())
+
+    # Here are the pound defines that HConfig.h would usually provide;
+    # We provide an empty HConfig.h file and do the defs and undefs
+    # here:
+    TODAY_DATE = datetime.today().strftime('%Y-%m-%d')
+    # ('OPENMP', None), ?
+    DEFINE_MACROS = [
+        ('CMAKE_BUILD_TYPE', '"Release"'),
+        ('HiGHSRELEASE', None),
+        ('IPX_ON', 'ON'),
+        ('HIGHS_GITHASH', '"%s"' % GITHASH),
+        ('HIGHS_COMPILATION_DATE', '"' + TODAY_DATE + '"'),
+        ('HIGHS_VERSION_MAJOR', HIGHS_VERSION_MAJOR),
+        ('HIGHS_VERSION_MINOR', HIGHS_VERSION_MINOR),
+        ('HIGHS_VERSION_PATCH', HIGHS_VERSION_PATCH),
+        ('HIGHS_DIR', '"' + HIGHS_DIR + '"'),
+    ]
+    UNDEF_MACROS = [
+        'OPENMP',  # unconditionally disable openmp
+        'EXT_PRESOLVE',
+        'SCIP_DEV',
+        'HiGHSDEV',
+        'OSI_FOUND',
+    ]
+
+    # Compile BASICLU as a static library to appease clang:
+    # (won't allow -std=c++11/14 option for C sources)
+    basiclu_sources = _get_sources('src/CMakeLists.txt',
+                                   'set(basiclu_sources\n', ')')
+    config.add_library(
+        'basiclu',
+        sources=basiclu_sources,
+        include_dirs=[
+            'src/',
+            'src/ipm/basiclu/include/',
+        ],
+        language='c',
+        macros=DEFINE_MACROS,
+    )
+
+    # highs_wrapper:
+    ipx_sources = _get_sources('src/CMakeLists.txt', 'set(ipx_sources\n', ')')
+    highs_sources = _get_sources('src/CMakeLists.txt', 'set(sources\n', ')')
+    ext = config.add_extension(
+        'highs_wrapper',
+        sources=['pyHiGHS/src/highs_wrapper.cxx'] + highs_sources + ipx_sources,
+        include_dirs=[
+
+            # highs_wrapper
+            'pyHiGHS/src/',
+            'src/',
+            'src/lp_data/',
+
+            # highs
+            'src/',
+            'src/io/',
+            'src/ipm/ipx/include/',
+
+            # IPX
+            'src/ipm/ipx/include/',
+            'src/ipm/basiclu/include/',
+        ],
+        language='c++',
+        libraries=['basiclu'],
+        define_macros=DEFINE_MACROS,
+        undef_macros=UNDEF_MACROS,
+    )
+    # Add c++11/14 support:
+    ext._pre_build_hook = pre_build_hook
+
+    # wrapper around HiGHS writeMPS:
```

Comment:
Question (PR is a bit large to hunt for the answer): this isn't exposed publicly right? It's useful functionality, but if we'd expose it, it should go in `scipy.io`.

Reply:
Not exposed now, but planned.